go-getting-started/controllers/insert_test.go

```diff
@@ -69,40 +62,42 @@ func TestControllers_InsertData(t *testing.T) {
 }
 
 func TestControllers_InsertDataID(t *testing.T) {
+
+	n := 5
+	b := make([]byte, n)
+	if _, err := rand.Read(b); err != nil {
+		panic(err)
+	}
+	randomID := fmt.Sprintf("%X", b)
+
 	tt := []struct {
 		name   string
 		want   models.Joke
 		insert models.Joke
+		id     string
 	}{
 		{
-			name: "Big number",
+			name: "id",
 			insert: models.Joke{
-				Body:  "asdada",
-				ID:    "sdfadsa",
+				Body:  "A Sunday school teacher is concerned that his students might be a litt",
+				ID:    randomID,
 				Score: 4,
-				Title: "asd",
-			},
-			want: models.Joke{
-				Body:  "",
-				ID:    "",
-				Score: 0,
-				Title: "",
+				Title: "his hand",
 			},
 		},
 	}
 
 	for _, tc := range tt {
 		t.Run(tc.name, func(t *testing.T) {
-			var joke models.Joke
 
-			collection := config.ConnectDB("mongodb+srv://jokesdb:jokesdb@joke.kxki9.mongodb.net/myFirstDatabase?retryWrites=true&w=majority")
+			collection := config.ConnectDB("mongodb://localhost:27017")
```

Comment:
You don't have to connect to the database in each iteration of the loop. Do it once before the loops is entered.